database/src/pool/postgres.rs

```diff
@@ -1608,6 +1630,34 @@ where
             .collect();
         Ok(requests)
     }
+
+    async fn enqueue_benchmark_job(&self, benchmark_job: &BenchmarkJob) -> anyhow::Result<()> {
```

Comment:
This is always tricky with these we-have-ORM-at-home situations, but it shouldn't really be possible to ever insert a job that is not in the queued state into the DB. I wonder if we should just pass the individual fields arguments (tag, target, backend, profile, benchmark_set) to this function, or wrap them in some other struct. That would also allow us to remove the `BenchmarkJob::new` constructor.